### PR TITLE
RET-4140

### PR DESCRIPTION
**Pull request by GauravChawlaMOJ**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-4140

### Change description ###

Pre-Hearing deposit callback implementation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/prehearingdeposit/PreHearingDepositController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            @RequestHeader("Authorization") String userToken,
+            @RequestBody CCDRequest ccdRequest) {
+        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+            return ResponseEntity.status(FORBIDDEN.value()).build();
+        }
+        PreHearingDepositData preHearingDepositData = ccdRequest.getCaseDetails().getPreHearingDepositData();
+        try {
+            preHearingDepositService.importData(preHearingDepositData, userToken);
+        } catch (IOException e) {
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Unable to import staff", e);
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/admin/prehearingdeposit/PreHearingDepositController.java:55]: change staff to pre hearing deposit data

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/domain/prehearingdeposit/PreHearingDepositData.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+public class PreHearingDepositData {
+    @JsonProperty("preHearingDepositImportFile")
+    private ImportFile preHearingDepositImportFile;
+    @JsonProperty("payeeName")
+    private String payeeName;
+
+    // For adding judge
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/domain/prehearingdeposit/PreHearingDepositData.java:16]: maybe change this comment

**Review by furnivall — APPROVED** [review]

File: src/test/resources/PreHearingDepositData.xlsx
Comment (hsjhita) [inline src/test/resources/PreHearingDepositData.xlsx:1]: Can this file be removed or the data changed please? This is live data and should not be committed to Git

**Review by hsjhita — CHANGES_REQUESTED** [review]

File: src/test/resources/PreHearingDepositData.xlsx
Comment (JustMehmet) [reply src/test/resources/PreHearingDepositData.xlsx:1]: Changed file content. Thanks Harpreet,

**Pull request merged**